hpeOneView/resources/security/users.py

```diff
@@ -108,8 +108,147 @@ def get_by_userName(self, userName):
             uri = self.URI + '/role/' + userName
             self._helper.validate_resource_uri(uri)
             data = self._helper.do_get(uri)
-            new_resource = self.new(self._connection, data)
-        else:
-            new_resource = None
+            result = data['members']
+            return result
+
+    def get_by_userName(self, name):
+        """
+        Gets a complete json body for username
+
+        Args:
+          name: userName of the user
+
+        Returns:
+           dict: User
+        """
+
+        uri = self.URI + '/' + name
```

Comment:
use this method self._helper.build_uri(name) instead of using '/'. This will append the given path to base uri.